pinot-broker/src/main/java/org/apache/pinot/broker/routing/BrokerRoutingManager.java

```diff
@@ -115,6 +115,7 @@ public class BrokerRoutingManager implements RoutingManager, ClusterChangeHandle
   private ZkHelixPropertyStore<ZNRecord> _propertyStore;
 
   private Set<String> _routableServers;
+  private Set<String> _disabledTables = new HashSet<>();
```

Comment:
This is not thread safe. I think an easier way to handle this is to add `_disabled` field into `RoutingEntry`, and update its value in `onAssignmentChange()`